### PR TITLE
Fix for issue 21.

### LINKED ISSUE #21: div_by_zero in atmos_ocean_fluxes_calc
I'm attempting a run through with Lori's xml, hitting a div_by_zero in this section of code which is calculating gas fluxes. https://github.com/NOAA-GFDL/FMScoupler/blob/f48e67ba9c343152045182e80df5c14021130d47/full/atmos_ocean_fluxes_calc.f90#L153 It would appear that the incoming ice gas fields are zero (this is the calling routine in atm_land_ice_flux_exchange): https://github.com/NOAA-GFDL/FMScoupler/blob/f48e67ba9c343152045182e80df5c14021130d47/full/atm_land_ice_flux_exchange.F90#L1235 I realize these values may be overridden at some point later on in the code (and hence the error probably doesn't even show up in a prod run), but should there be something here that at least warns of the div by zero, or perhaps provides a default flux value in the event of 0 fields?

### PR DESCRIPTION
**Pull request by GFDL-Eric**
Added epsilons to places where a potential div by zero can occur. Fixes #21

### CODE REVIEW

File: full/atmos_ocean_fluxes_calc.f90 [hunk 1]
```diff
@@ -117,7 +117,7 @@ subroutine atmos_ocean_fluxes_calc(gas_fields_atm, gas_fields_ice,&
                     & gas_fields_atm%bc(n)%field(ind_u10)%values(i) /&
                     & (770.+45.*gas_fluxes%bc(n)%param(1)**(1./3.)) *&
                     & 101325./(rdgas*wtmair*1e-3*tsurf(i) *&
-                    & gas_fields_ice%bc(n)%field(ind_alpha)%values(i))
+                    & max(gas_fields_ice%bc(n)%field(ind_alpha)%values(i),epsln))
```
Comment (underwoo) [inline full/atmos_ocean_fluxes_calc.f90]: @GFDL-Eric is there a guarantee that `gas_fields_ice%bc(n)%field(ind_alpha)%values(i)` is always greater than or equal to zero?

Comment (GFDL-Eric) [reply full/atmos_ocean_fluxes_calc.f90]: I don't believe there's a check for non-negative values anywhere, on the other hand I'm not quite sure what a concentration value less than zero would physically represent.

**Comment by thomas-robinson** [comment]
Is this going in for 2020.01?

**Comment by underwoo** [comment]
My vote is no.  The fms2_io updates are taking longer than expected to
correct.  I'd rather not add this to the mix.  We can add this with the
Nesting updates, or in the third 2020 release.
--
Seth Underwood

S/W Development Process Lead
Modeling Systems Group
GFDL/NOAA/DOC
201 Forrestal Road
Princeton, NJ 08540-6649

(609) 452-5847 Office
(304) 376-9002 Cell
(609) 987-5063 Fax
Seth.Underwood@noaa.gov

**Comment by thomas-robinson** [comment]
@GFDL-Eric how extensively was this tested?  Is it possible for this to change answers?

**Comment by GFDL-Eric** [comment]
I was mostly doing testing within the test_data_override suite to identify the issue, and found that there were answer changes in previous incarnations when I was just adding an epsilon value instead of doing the max function, but with the max function and the epsilon sufficiently small (1e-30), I don't see answer changes in the test_data_override suite or in my test with a land xml (lm4p1reg.xml). Those tests were a while ago and obviously didn't include anything like AM4.1 or ESM, since, well, I haven't had much success getting those to run at all.

**Comment by thomas-robinson** [comment]
I ran this through the AM4 suite and it passed in prod mode.  It should be good for more extensive testing

**Pull request merged**